### PR TITLE
Support nondivisible alignment times in `$find`

### PR DESCRIPTION
**Pull request by noahsilas**
When using `alignmentInterval` that doesn't evenly divide the hour, you can end up with surprising results. For example, when using an alignment like "90 minutes", the grid alignment depended on the availability window that was being searched. This meant that searching for appointments on a single day could return slots aligned to two distinct grids as your search `start` time changed.

To resolve this, we need to choose a standard grid to use for this alignment, and ensure that we are matching it when generating appointment times in $find. In this PR we choose to align a daily grid to midnight (potentially offset by `alignmentOffset` scheduling parameter).

One tricky aspect to this is that the alignment grid should shift with local timezone rules. For example, imagine you are using a 90 minute alignment interval and your slots start at 9am PDT (-7). After the DST transition, if the alignment grid does not also shift, your slots would start beginning at 10am PST (-8).

Most businesses don't change their hours on these transitions, and so when they configure their alignment, the intention is usually for it to be consistent: "My day starts at 9am PT, and my first slot should align with that." To that end, we introduce a new SchedulingParameter attribute, `alignmentTimezone`. This lets you choose the zone that your alignment grid is anchored with respect to.

## Backwards compatibility note:
In cases where your slots are divisible by one hour, this does not make a difference[^1] (10,20,15,30,60 minute slots all evenly fill the entire daily grid). This was the only officially supported scenario in our previous alpha implementation; folks using other alignment values may see their scheduling grid suddenly jump when this is released.

I'm setting the default alignmentTimezone value to `Etc/UTC` - this is an easy-to-explain default that works well for some places (Arizona, Hawaii, Japan, anywhere that doesn't observe DST). It is likely that most users will want to override this to match their local timezone (e.g., `America/New_York`, `America/Los_Angeles`, `Europe/Paris`).

## Future Related Changes:
- This is building towards a solution for #8734, in which we would change the `alignmentInterval` default value. The most requested default value is one that matches the scheduling duration, allowing tip-to-tail booking by default. In that world, we can almost guarantee that alignment grids will diverge from evenly divisible values ("90 minutes" being a common example). For that reason, some kind of change like this is a prerequisite.

- With this updated slot generation in place, we should revisit enforcing alignment in $book. The original implementation assumed that the input value was evenly divisible, which broke on some outputs from $find in these non-divisible alignment values. We removed that restriction in #8331, which means that currently $book can be used to create appointments that violate scheduling constraints. We should update $book/$hold to test alignment with these new rules.

[^1]: Okay, it makes a difference in some rare edge cases, such as when you are on Lord Howe Island on a day in which a Daylight Savings Time transition shifts the clock by thirty minutes. I suspect that scheduling discontinuities in such environments are unsurprising.

### CODE REVIEW

**Comment by noahsilas** [comment]
@claude please review

File: packages/server/src/fhir/operations/utils/scheduling-parameters.ts [hunk 1]
```diff
@@ -159,6 +168,19 @@ function durationToMinutes(extension: WithPath<Extension>): number {
...
 }
 
+function assertValidTimezone(ext: WithPath<Extension>): void {
+  assertExtensionCode(ext);
+  // Check that we can build a Temporal.ZonedDateTime with the given timezone.
+  // Note that this accepts non-canonical timezone identifiers (example:
+  // "US/Pacific" is an alias for "America/Los_Angeles"), and is
+  // case-insensitive (we accept "america/los_angeles" as valid).
+  try {
+    Temporal.Instant.fromEpochMilliseconds(0).toZonedDateTimeISO(ext.valueCode);
```
Comment (mattwiller) [inline packages/server/src/fhir/operations/utils/scheduling-parameters.ts]: Technically, I think you could avoid re-creating `Temporal.Instant.fromEpochMilliseconds(0)` each time and could re-use a constant defined outside the function body to make this slightly more performant

**Review by mattwiller — APPROVED** [review]

File: packages/server/src/fhir/operations/utils/scheduling-parameters.ts (hunk 1 above)
Comment (noahsilas) [reply packages/server/src/fhir/operations/utils/scheduling-parameters.ts]: Nice catch, thanks!
